lib/sinon/behavior.js

```diff
@@ -171,7 +173,12 @@ var proto = {
             return (this.promiseLibrary || Promise).reject(this.returnValue);
         } else if (this.callsThrough) {
             var wrappedMethod = this.effectiveWrappedMethod();
+
             return wrappedMethod.apply(context, args);
+        } else if (this.callsThroughWithNew) {
+            var WrappedClass = this.effectiveWrappedMethod();
+            var F = WrappedClass.bind.apply(WrappedClass, concat([null], flat(args)));
```

Comment:
Why are you flattening the arguments array? That would change the parameters sent to the class, like 
`new Stub([a,b,c],[1,2,3])` would end up calling `new Original(a,b,c,1,2,3)`, which is not the same at all.

Reply:
The arguments object is treated as a single entity when concatenated with any other array, so it needs to be converted into a normal array.  You're right, though, flat is not the right tool here.  Changing to slice.  Resolved in 8e00c22.